src/rpc/mempool.cpp

```diff
@@ -242,6 +243,7 @@ static RPCHelpMan testmempoolaccept()
                         result_inner.pushKV("reject-reason", "missing-inputs");
                     } else {
                         result_inner.pushKV("reject-reason", state.GetRejectReason());
+                        if (!state.GetDebugMessage().empty()) result_inner.pushKV("reject-details", state.GetDebugMessage());
```

Comment:
A simpler way to do this might just be `ToString()`.

It ensures the field exists as long as reject-reason exists. That'd be a simpler interface to understand, as the current one depends on very specific internal stuff - I can't really tell you off the top of my head where we have/don't add a debug string.

It does mean the (very short) first part of the error string is repeated in both places, but perhaps that is convenient for users that want the more verbose one.

Also nit, I heard underscores are better for json parsers
```suggestion
                        result_inner.pushKV("reject_details", state.ToString());
```

Reply:
Sure I'll use `ToString()` here. Going to leave the hyphens though because there are several other hyphenated fields in the response object already... but I do see lots of underscores in other RPC responses 😬

Reply:
So actually, after making this change I discovered several tests fail everywhere `reject-reason` is checked. There will be a lot of diffs like this... let me know if this changes your mind at all about that `if` condition there.

```diff
--- a/test/functional/mempool_accept.py
+++ b/test/functional/mempool_accept.py
@@ -109,7 +109,7 @@ class MempoolAcceptanceTest(BitcoinTestFramework):
         ))
         # ... 0.99 passes
         self.check_mempool_result(
-            result_expected=[{'txid': txid_in_block, 'allowed': False, 'reject-reason': 'txn-already-known'}],
+            result_expected=[{'txid': txid_in_block, 'allowed': False, 'reject-reason': 'txn-already-known', 'reject-details': 'txn-already-known'}],
             rawtxs=[raw_tx_in_block],
             maxfeerate=0.99,
         )
```

Reply:
@pinheadmz could pop that field just like the check_mempool_result pops fee field?